docs/rest-api.adoc

```diff
@@ -0,0 +1,189 @@
+        "hudson.model.Computer.Configure"
+    ],
+    "agentNames": [
+        "bar",
+        "baz"
+    ]
+}
+----
+
+== API methods for assigning a sid to a role
+
+=== `assignSidToGlobalRole`
+
+Assigns a sid to the role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/assignSidToGlobalRole`. The following query
+parameters are required:
+
+* `roleName`: The sid will be assigned to the global role with the name equal
+to this parameter.
+* `sid`: The sid to be assigned to the role with the name equal to the value of
+`roleName`.
+
+Using `curl`, for example, a sid "foo" can be assigned to the role "bar"
+
+[source,bash]
+----
+curl -X POST -d 'roleName=bar&sid=foo' \
+    http://localhost:8080/folder-auth/assignSidToGlobalRole
+----
+
+=== `assignSidToFolderRole`
+
+Assigns a sid to the role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/assignSidToFolderRole`. The following query
+parameters are required:
+
+* `roleName`: The sid will be assigned to the folder role with the name equal
+to this parameter.
+* `sid`: The sid to be assigned to the role with the name equal to the value of
+`roleName`.
+
+Using `curl`, for example, a sid "foo" can be assigned to the role "bar"
+
+[source,bash]
+----
+curl -X POST -d 'roleName=bar&sid=foo' \
+    http://localhost:8080/folder-auth/assignSidToFolderRole
+----
+
+=== `assignSidToAgentRole`
+
+Assigns a sid to the role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/assignSidToAgentRole`. The following query
+parameters are required:
+
+* `roleName`: The sid will be assigned to the agent role with the name equal
+to this parameter.
+* `sid`: The sid to be assigned to the role with the name equal to the value of
+`roleName`.
+
+Using `curl`, for example, a sid "foo" can be assigned to the role "bar"
+
+[source,bash]
+----
+curl -X POST -d 'roleName=bar&sid=foo' \
+    http://localhost:8080/folder-auth/assignSidToGlobalRole
+----
+
+== API methods for deleting a role
+
+=== `deleteGlobalRole`
+
+Deletes a global role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/deleteGlobalRole`. The query parameter
+`roleName` is required.
+
+Using `curl`, for example, a role with name "foo" can be deleted
+
+[source,bash]
+----
+curl -X POST -d 'roleName=foo' http://localhost:8080/folder-auth/deleteGlobalRole
+----
+
+=== `deleteFolderRole`
+
+Deletes a folder role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/deleteGlobalRole`. The parameter
+`roleName` is required.
+
+[source,bash]
+----
+curl -X POST -d 'roleName=foo' http://localhost:8080/folder-auth/deleteFolderRole
+----
+
+=== `deleteAgentRole`
+
+Deletes an agent role identified by its name. Requires POST to
+`${JENKINS_URL}/folder-auth/deleteGlobalRole`. The parameter
+`roleName` is required.
+
+[source,bash]
+----
+curl -X POST -d 'roleName=foo' http://localhost:8080/folder-auth/deleteAgentRole
+----
+
+== Logging in to Jenkins
```

Comment:
It is better to use API tokens for credentials.
Also, a token will be needed on instances with CSRF protection
https://wiki.jenkins.io/display/JENKINS/Remote+access+AP